cpp/src/join/distinct_hash_join.cu

```diff
@@ -181,8 +181,12 @@ distinct_hash_join<HasNested>::inner_join(rmm::cuda_stream_view stream,
   auto const probe_indices_begin =
     thrust::make_transform_output_iterator(probe_indices->begin(), output_fn{});
 
-  auto const [probe_indices_end, _] = this->_hash_table.retrieve(
-    iter, iter + probe_table_num_rows, probe_indices_begin, build_indices_begin, stream.value());
+  auto const [probe_indices_end, _] =
+    this->_hash_table.retrieve(iter,
+                               iter + probe_table_num_rows,
+                               probe_indices_begin,
+                               build_indices_begin,
+                               cuco::cuda_stream_ref{stream.value()});
```

Comment:
question: is the explicit `cuco::cuda_stream_ref` construction necessary?

Reply:
Yes, otherwise the compiler thinks we're trying to call the other overload of this function.

Reply:
Thanks. Can you please check my other review suggestion? That can avoid a breaking change with the upcoming cuco refactoring. The PR should be ready to go once @miscco's and my reviews are addressed.